### DISCUSSION TITLE
Flatten: Using a single space character as delimiter

### DISCUSSION THREAD

**Discussion by raufaser** [flwyd/adif-multitool#10] [General]
Hi.

I am trying to flatten my QSOs. I put multiple World Castle Award references into the field comment. Simply seperated by a whitespace. 

I am using Windows Power Shell and something like this doesn't work:

`.\adifmt-windows-amd64.exe flatten --fields comment --delimiter "comment=' '"`

And this:

`.\adifmt-windows-amd64.exe flatten --fields comment --delimiter 'comment= '`

gives an error:

`Error running flatten: don't know delimiter for field "COMMENT" of type String`

Any idea how to get it running?

**Comment by flwyd** [comment]
Great catch, I didn't think of testing with whitespace as a delimiter.  I've filed this as issue #11.

Until that's fixed, the only way I can think to solve your problem is to use a text editor or something like `sed` or `awk` to replace your space delimiters with commas or something.  I should be able to push a fix soon.

**Comment by flwyd** [comment]
Fixed in v0.1.15.  I actually did have a test for space as a delimiter, but that test didn't go through the flag parsing layer, so I've added some end-to-end tests in the `adifmt/testdata` directory.
